src/Microsoft.PowerShell.Commands.Utility/commands/utility/MatchString.cs

```diff
@@ -221,12 +265,43 @@ public override string ToString()
         public string ToString(string directory)
         {
             string displayPath = (directory != null) ? RelativePath(directory) : _path;
+            string modifiedLine;
+            if (Emphasize)
+            {
+                string open;
+                string close;
+
+                if (_supportsVirtualTerminal)
+                {
+                    open = "\u001b[31m";
+                    close = "\u001b[0m";
+                }
+                else
+                {
+                    open = "*";
+                    close = "*";
+                }
+
+                StringBuilder sb = new StringBuilder(Line);
+                for (int i = 0; i < _matchIndexes.Count; i++)
+                {
+                    int offset = open.Length + close.Length;
+                    sb.Insert(_matchIndexes[i] + (i * offset), open);
+                    sb.Insert(_matchIndexes[i] + _matchLengths[i] + ((i * offset) + open.Length), close);
```

Comment:
The algorithm is very expensive. I'd use String.Create() because we can calculate target string length.

Reply:
Done.